### PR TITLE
Add Stress and Longhaul tests for Windows IoT Core x64

### PR DESCRIPTION
**Pull request by SlavyMihov**

### CODE REVIEW

File: builds/e2e/longhaul.yaml [hunk 1]
```diff
@@ -349,4 +349,128 @@ jobs:
...
+          
+################################################################################
+  - job: setup_iotuap_amd64_moby
+################################################################################
+    displayName: Windows IoT Core AMD64 Moby (Setup Only)
+    condition: eq(variables['run.iotuap.amd64.moby'], 'true')
+    pool:
+      name: $(pool.name)
+      demands:
+        - longhaul_iotuap_x64 -equals true
```
Comment (philipktlin) [inline builds/e2e/longhaul.yaml]: Please work with Chanisha; to follow the convention she created; and try to reuse template she is going to create.  You can refer to this PR, https://github.com/Azure/iotedge/pull/1296.

File: builds/e2e/longhaul.yaml [hunk 2]
```diff
@@ -349,4 +349,128 @@ jobs:
...
+          snitch.storage.account: '$(snitch.storage.account)'
+          snitch.storage.masterKey: '$(StorageAccountMasterKeyStress)'
+          loadGen.message.frequency: '$(loadGen.message.frequency.amd64)'
+          
+################################################################################
+  - job: iotuap_amd64_moby
+################################################################################
+    displayName: Windows IoT Core AMD64 Moby
+    dependsOn: setup_iotuap_amd64_moby
+    condition: eq(variables['run.iotuap.amd64.moby'], 'true')
```
Comment (philipktlin) [inline builds/e2e/longhaul.yaml]: Chanisha used the same variable 'run.windows.amd64.moby'; I am ok, either use different variable for different windows platform or use the same variable for same bits,e.g. 64 bits.  Please align with her.

File: builds/e2e/longhaul.yaml (hunk 1 above)
Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: I rebased my branch to include Chanisha's changes after merged to Azure/iotedge.

File: builds/e2e/longhaul.yaml (hunk 2 above)
Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: Synced up with Chanisha and both decided on variables that are distinct.

File: builds/e2e/longhaul.yaml [hunk 3]
```diff
@@ -346,4 +346,128 @@ jobs:
...
+    steps:
+      - checkout: none
+      - task: AzureKeyVault@1
+        displayName: 'Azure Key Vault'
+        inputs:
+          azureSubscription: $(azure.subscription)
+          KeyVaultName: $(azure.keyVault)
+          SecretsFilter: 'edgebuilds-azurecr-io-username,edgebuilds-azurecr-io-pwd,IotHubStressConnString,EventHubStressConnStr,StorageAccountMasterKeyStress,SnitchLongHaulAlertUrl'
+      - task: DownloadBuildArtifacts@0
+        displayName: 'Download Edgelet Artifacts'
```
Comment (philipktlin) [inline builds/e2e/longhaul.yaml]: Please work with Chanisha to refactor download steps as template for reuse.  I tried to put azure key vault step into template before, but I remember this may cause some credential issue, you can try on yourself to see if it works now.

Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: Let’s work on a template for reuse across all Windows editions, but in a separate code review and keep this one focused on Windows IoT Core - adding the missing Stress and Longhaul tests to get on par with Windows Enterprise and Windows Server (all x64).

File: builds/e2e/longhaul.yaml [hunk 4]
```diff
@@ -346,4 +346,128 @@ jobs:
...
+          
+################################################################################
+  - job: setup_iotuap_amd64_moby
+################################################################################
+    displayName: Windows IoT Core AMD64 Moby (Setup Only)
+    condition: eq(variables['run.iotuap.amd64.moby'], 'true')
+    pool:
+      name: $(pool.name)
+      demands:
+        - longhaul_iotuap_x64 -equals true
```
Comment (philipktlin) [inline builds/e2e/longhaul.yaml]: please refer to this OneNote page, https://microsoft.sharepoint.com/teams/Azure_IoT/_layouts/OneNote.aspx?id=%2Fteams%2FAzure_IoT%2FShared%20Documents%2FEngineering%2FIoTHubNotes&wd=target%28Azure%20IoT%20Edge%2FWiki.one%7C70253E27-7EC5-43AD-89E4-C9BA3943D8CF%2FIoT%20Edge%20VMs%5C%2FDevices%7C0330C378-E885-4582-A9ED-C80CBEA3F05F%2F%29
onenote:https://microsoft.sharepoint.com/teams/Azure_IoT/Shared%20Documents/Engineering/IoTHubNotes/Azure%20IoT%20Edge/Wiki.one#IoT%20Edge%20VMs/Devices&section-id={70253E27-7EC5-43AD-89E4-C9BA3943D8CF}&page-id={0330C378-E885-4582-A9ED-C80CBEA3F05F}&end and add/update for newly created machines.

Currently we use "run-long-haul" capability to indicate that machine is used for long haul with other parameters e.g. 
agent-os-name to distinguish for different platforms.

File: builds/e2e/longhaul.yaml [hunk 5]
```diff
@@ -346,4 +346,128 @@ jobs:
...
+################################################################################
+  - job: iotuap_amd64_moby
+################################################################################
+    displayName: Windows IoT Core AMD64 Moby
+    dependsOn: setup_iotuap_amd64_moby
+    condition: eq(variables['run.iotuap.amd64.moby'], 'true')
+    pool:
+      name: $(pool.name)
+      demands:
+        - longhaul_iotuap_x64 -equals true
```
Comment (philipktlin) [inline builds/e2e/longhaul.yaml]: Please refer to above comment.

File: builds/e2e/templates/deploy-iotuap-setup.yaml [hunk 6]
```diff
@@ -0,0 +1,66 @@
+parameters:
+  release.label: ''
```
Comment (philipktlin) [inline builds/e2e/templates/deploy-iotuap-setup.yaml]: for those parameter it is not used in this template, please remove them.

File: builds/e2e/longhaul.yaml [hunk 7]
```diff
@@ -346,4 +346,128 @@ jobs:
           snitch.storage.account: '$(snitch.storage.account)'
           snitch.storage.masterKey: '$(StorageAccountMasterKeyStress)'
           loadGen.message.frequency: '$(loadGen.message.frequency.amd64)'
-         
\ No newline at end of file
+          
+################################################################################
+  - job: setup_iotuap_amd64_moby
+################################################################################
+    displayName: Windows IoT Core AMD64 Moby (Setup Only)
```
Comment (philipktlin) [inline builds/e2e/longhaul.yaml]: Just want to confirm, the reason we need to split into 2 jobs: setup and test.  Because we can't install x64 version of VSTS agent on Win IoT Core x64; is it right?  why it fails?

File: builds/e2e/longhaul.yaml (hunk 4 above)
Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: What is the point of using two parameters:
 * run-long-haul=**true**
 * agent-os-name=**WinIoTCore_x64**

instead of just one (that carries information about both the test type and the OS type)?
 * longhaul_iotuap_x64=**true**

File: builds/e2e/longhaul.yaml [hunk 8]
```diff
@@ -346,4 +346,128 @@ jobs:
...
+    pool:
+      name: $(pool.name)
+      demands:
+        - longhaul_iotuap_x64 -equals true
+    variables:
+      edgelet.artifact.name: 'iotedged-windows'
+    steps:
+      - checkout: none
+      - task: AzureKeyVault@1
+        displayName: 'Azure Key Vault'
```
Comment (philipktlin) [inline builds/e2e/longhaul.yaml]: you don't need this key vault step in this setup job.

File: builds/e2e/longhaul.yaml (hunk 7 above)
Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: Yes, we need a separate _Windows IoT Core AMD64 Moby (Setup Only)_ stage to install/update the iotedge service on the runner VM (Windows IoT Core x64), which causes a reboot.
Then we have the other stage (_Windows IoT Core AMD64 Moby_) that runs the Longhaul test bypassing installation of iotedge service.

Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: Same situation with Stress test - we need two stages to handle the reboot of the runner VM.

Comment (philipktlin) [reply builds/e2e/longhaul.yaml]: Question: does "Setup only" step restart the machine?

Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: Yes, installation/update of the iotedge service on Windows IoT Core triggers a reboot (just like with the E2E pipeline).

File: builds/e2e/longhaul.yaml (hunk 8 above)
Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: I think we need that, since we need to get the edgebuilds-azurecr-io username and password used in the last step of this stage:
_- template: templates/deploy-iotuap-setup.yaml_

Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: I take that back, not being used. I'll remove that step.

File: builds/e2e/longhaul.yaml [hunk 9]
```diff
@@ -346,4 +346,128 @@ jobs:
...
+        inputs:
+          azureSubscription: $(azure.subscription)
+          KeyVaultName: $(azure.keyVault)
+          SecretsFilter: 'edgebuilds-azurecr-io-username,edgebuilds-azurecr-io-pwd,IotHubStressConnString,EventHubStressConnStr,StorageAccountMasterKeyStress,SnitchLongHaulAlertUrl'
+      - task: DownloadBuildArtifacts@0
```
Comment (philipktlin) [inline builds/e2e/longhaul.yaml]: you don't need to download artifacts again; since this should be the same machine we picked to run long haul test.

File: builds/e2e/templates/deploy-iotuap-setup.yaml [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    displayName: 'Update iotedged'
+    inputs:
+      Machines: '$(iotuap-x64-machine)'
+      UserName: '$(iotuap-x64-login)'
+      UserPassword: '$(iotuap-x64-password)'
+      InlineScript: |
+        # Check iotedge version
+        $serviceName = 'iotedge'
+        
+        If (Get-Service $serviceName -ErrorAction SilentlyContinue) {
```
Comment (philipktlin) [inline builds/e2e/templates/deploy-iotuap-setup.yaml]: please wrap this script block in existing E2E test script.

File: builds/e2e/templates/deploy-iotuap-setup.yaml [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+      SourceFolder: "$(Build.StagingDirectory)/${{ parameters['images.artifact.name'] }}"
+      TargetFolder: "\\\\$(iotuap-x64-machine)\\c$\\Data\\e2etests\\artifacts\\${{ parameters['images.artifact.name'] }}"
+      CleanTargetFolder: true
+  - task: PowerShellOnTargetMachines@3
+    displayName: 'Update iotedged'
+    inputs:
+      Machines: '$(iotuap-x64-machine)'
+      UserName: '$(iotuap-x64-login)'
+      UserPassword: '$(iotuap-x64-password)'
+      InlineScript: |
```
Comment (philipktlin) [inline builds/e2e/templates/deploy-iotuap-setup.yaml]: please wrap this inline script in existing E2E test script.

File: builds/e2e/templates/longhaul-deploy-iotuap.yaml [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+      TargetFolder: "\\\\$(iotuap-x64-machine)\\c$\\Data\\e2etests\\artifacts\\${{ parameters['edgelet.artifact.name'] }}"
+      CleanTargetFolder: true
+  - task: CopyFiles@2
+    displayName: 'Copy Images Artifact'
+    inputs:
+      SourceFolder: "$(Build.StagingDirectory)/${{ parameters['images.artifact.name'] }}"
+      TargetFolder: "\\\\$(iotuap-x64-machine)\\c$\\Data\\e2etests\\artifacts\\${{ parameters['images.artifact.name'] }}"
+      CleanTargetFolder: true
+  - task: PowerShellOnTargetMachines@3
+    displayName: 'Run Long Haul Deployment'
```
Comment (philipktlin) [inline builds/e2e/templates/longhaul-deploy-iotuap.yaml]:
```suggestion
    displayName: 'Run Long Haul Deployment on $(iotuap-x64-machine)'
```

Comment (philipktlin) [reply builds/e2e/templates/longhaul-deploy-iotuap.yaml]: please try this suggestion, if it is possible to use variable in displayName field.

File: builds/e2e/longhaul.yaml (hunk 9 above)
Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: Makes sense. Removing these steps.

File: builds/e2e/templates/stresstest-deploy-iotuap.yaml [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+      CommunicationProtocol: Http
+    condition: succeededOrFailed()
```
Comment (philipktlin) [inline builds/e2e/templates/stresstest-deploy-iotuap.yaml]: what is this used for?

File: builds/e2e/templates/deploy-iotuap-setup.yaml (hunk 6 above)
Comment (SlavyMihov) [reply builds/e2e/templates/deploy-iotuap-setup.yaml]: Removed.

File: builds/e2e/templates/longhaul-deploy-iotuap.yaml (hunk 12 above)
Comment (SlavyMihov) [reply builds/e2e/templates/longhaul-deploy-iotuap.yaml]: Good suggestion.
Done.

File: builds/e2e/templates/stresstest-deploy-iotuap.yaml (hunk 13 above)
Comment (SlavyMihov) [reply builds/e2e/templates/stresstest-deploy-iotuap.yaml]: Removing this as it is not being used here.

File: builds/e2e/templates/deploy-iotuap-setup.yaml (hunk 11 above)
Comment (SlavyMihov) [reply builds/e2e/templates/deploy-iotuap-setup.yaml]: I do not think we have an existing E2E test script. Everything for E2E is done as inline PS commands in the stages.
We can do that (wrap all inline scripts in one test script) as a separate pull request, to not over-complicate this one.

File: builds/e2e/longhaul.yaml (hunk 4 above)
Comment (philipktlin) [reply builds/e2e/longhaul.yaml]: It is easier to understand and keep consistent with existing tests.

File: builds/e2e/longhaul.yaml (hunk 7 above)
Comment (philipktlin) [reply builds/e2e/longhaul.yaml]: please put a comment in setup template file to mention about auto-reboot of the device after installation.  say iotedge-win7 run setup only step and iotedge-iotuap7 restarted; then the next job picked up by iotedge-win7 immediately and run.  is it possible that iotedge-iotuap7 doesn't completely start up?  If that's the reason, we should be able to merge these 2 jobs into 1. right?

File: builds/e2e/templates/deploy-iotuap-setup.yaml (hunk 11 above)
Comment (philipktlin) [reply builds/e2e/templates/deploy-iotuap-setup.yaml]: Please check existing Windows x64; each step mainly call to RunE2ETests.ps1 only.  Therefore this inline script should move  into this script.

File: builds/e2e/templates/deploy-iotuap-setup.yaml [hunk 14]
```diff
@@ -0,0 +1,49 @@
+parameters:
+  edgelet.artifact.name: ''
+  images.artifact.name: ''
+
+steps:
+  - task: CopyFiles@2
+    displayName: 'Copy Edgelet Artifact'
+    inputs:
+      SourceFolder: "$(Build.StagingDirectory)/${{ parameters['edgelet.artifact.name'] }}"
+      TargetFolder: "\\\\$(iotuap-x64-machine)\\c$\\Data\\e2etests\\artifacts\\${{ parameters['edgelet.artifact.name'] }}"
```
Comment (philipktlin) [inline builds/e2e/templates/deploy-iotuap-setup.yaml]: do you need to create e2etests folder if it doesn't exist?

File: builds/e2e/longhaul.yaml (hunk 4 above)
Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: Keep in mind that Windows IoT Core is different - unlike the other test setups it needs two VMs (a VSTS agent VM and a runner VM). The runner VM needs to be Windows IoT Core, but the VSTS agent machine can be anything (Windows Pro/Server or Linux), it cannot be Windows IoT Core.
Therefore the condition _agent-os-name=**WinIoTCore_x64**_ won't make much sense.
The only thing we need to guarantee is that we have 1:1 mapping between the Longhaul test pipeline and its dedicated VSTS agent VM that targets the Windows IoT Core runner.

Let me know if you still would like to see this change.

File: builds/e2e/templates/deploy-iotuap-setup.yaml [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+                # shutdown -r
+            }
+        } Else {
+            Write-Host "$serviceName not found."
+            Write-Host "Installing $serviceName..."
+            . C:\Data\e2etests\artifacts\core-windows\scripts\windows\setup\IotEdgeSecurityDaemon.ps1
+            deploy-iotedge
+        }
+      CommunicationProtocol: Http
+      ErrorActionPreference: silentlyContinue
```
Comment (philipktlin) [inline builds/e2e/templates/deploy-iotuap-setup.yaml]: Why need this?

File: builds/e2e/templates/deploy-iotuap-setup.yaml (hunk 14 above)
Comment (SlavyMihov) [reply builds/e2e/templates/deploy-iotuap-setup.yaml]: I do not think this is needed. This is how it's been done for the other Windows tests. I have not seen any issues there. Perhaps the _CopyFiles_ task handles that differently than the _PowerShell_ task that would fail.

File: builds/e2e/templates/deploy-iotuap-setup.yaml [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+    condition: succeededOrFailed()
```
Comment (philipktlin) [inline builds/e2e/templates/deploy-iotuap-setup.yaml]: does this means this step will be run no matter previous steps are succeeded or failed? if yes, remove it. only run if all previous tasks are passed.

Comment (SlavyMihov) [reply builds/e2e/templates/deploy-iotuap-setup.yaml]: Removed.

File: builds/e2e/templates/deploy-iotuap-setup.yaml (hunk 15 above)
Comment (SlavyMihov) [reply builds/e2e/templates/deploy-iotuap-setup.yaml]: If the runner VM reboots too quickly before this step finishes, then this is perceived as if the step failed. Regardless, we still want to run the next stage _"Windows IoT Core AMD64 Moby"_ that runs the actual Stress/Longhaul test.

File: builds/e2e/longhaul.yaml (hunk 7 above)
Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: I was not able to handle the runner's reboot in one step. The step just fails if the device reboots. Therefore I came up with this two-step approach:
* the first step "Setup only" installs/updates the service and reboots the runner
* the second step picks up the rebooted device and runs the Longhaul test

If the runner VM does not completely start, the second step (that runs the test) will fail being unable to connect to the runner.

Added a comment explaining the two stages and the reboot.

File: builds/e2e/templates/deploy-iotuap-setup.yaml (hunk 11 above)
Comment (SlavyMihov) [reply builds/e2e/templates/deploy-iotuap-setup.yaml]: I moved the inline script to RunE2ETests.ps1, feedback is welcome

File: builds/e2e/longhaul.yaml (hunk 4 above)
Comment (philipktlin) [reply builds/e2e/longhaul.yaml]: Make sense that we should not use agent-os-name; instead we should name it as runner-os-name or target-os-name.

File: builds/e2e/longhaul.yaml (hunk 7 above)
Comment (philipktlin) [reply builds/e2e/longhaul.yaml]: "...the first step "Setup only" installs/updates the service and reboots the runner..."; would you please show me where the code did the reboot for runner device?

"...If the runner VM does not completely start, the second step (that runs the test) will fail being unable to connect to the runner...." does this happen sometimes? how about have a step to detect where runner device is restarted?

File: builds/e2e/templates/deploy-iotuap-setup.yaml [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+      CleanTargetFolder: true
+  - task: PowerShellOnTargetMachines@3
+    displayName: 'Update iotedged'
+    inputs:
+      Machines: '$(iotuap-x64-machine)'
+      UserName: '$(iotuap-x64-login)'
+      UserPassword: '$(iotuap-x64-password)'
+      InlineScript: |
+        #Requires -RunAsAdministrator
+        $testName="void"
```
Comment (philipktlin) [inline builds/e2e/templates/deploy-iotuap-setup.yaml]: what is "void" test name?

File: builds/e2e/templates/deploy-iotuap-setup.yaml [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+        $ArtifactInfo=convertfrom-stringdata (get-content C:\Data\e2etests\artifacts\${{ parameters['images.artifact.name'] }}\artifactInfo.txt -raw)
+        C:\Data\e2etests\artifacts\${{ parameters['images.artifact.name'] }}\scripts\windows\test\Run-E2ETest.ps1 -E2ETestFolder "C:\Data\e2etests" -ArtifactImageBuildNumber "$($ArtifactInfo.BuildNumber)" -TestName "$testName" -ReleaseLabel " " -ContainerRegistryUsername " " -ContainerRegistryPassword " " -IoTHubConnectionString " " -EventHubConnectionString " "
+      CommunicationProtocol: Http
+      ErrorActionPreference: silentlyContinue
```
Comment (philipktlin) [inline builds/e2e/templates/deploy-iotuap-setup.yaml]: why do we need this?

File: builds/e2e/templates/deploy-iotuap-setup.yaml (hunk 17 above)
Comment (SlavyMihov) [reply builds/e2e/templates/deploy-iotuap-setup.yaml]: Obsolete.

File: builds/e2e/templates/deploy-iotuap-setup.yaml (hunk 18 above)
Comment (SlavyMihov) [reply builds/e2e/templates/deploy-iotuap-setup.yaml]: Obsolete.

File: scripts/windows/test/Run-E2ETest.ps1 [hunk 19]
```diff
@@ -1319,6 +1300,8 @@ Else
     $BypassInstallationFlag = $null
 }
 
+TestSetup
```
Comment (philipktlin) [inline scripts/windows/test/Run-E2ETest.ps1]: This may break if test name is "All".  Please revert this change.

File: scripts/windows/test/Setup-Env.ps1 [hunk 20]
```diff
@@ -0,0 +1,112 @@
...
+Else    # Windows Pro/Enterprise/Server - uninstall iotedge
+{
+	Write-Host "Uninstall iotedged"
+	# no reboot
+	Uninstall-IoTEdge -Force
+}
+
+# hide exit error caused by target reboot
+# next step in the pipeline will verify if the device comes back successfully
+Exit 0
```
Comment (philipktlin) [inline scripts/windows/test/Setup-Env.ps1]: this should move inside "If ($osEdition -eq "IoTUAP")" block.

File: scripts/windows/test/Run-E2ETest.ps1 [hunk 21]
```diff
@@ -212,21 +212,6 @@ Function AppendInstallationOption([string] $testCommand)
     Return $testCommand += " -a `"$IoTEdgedWorkingFolder`""
 }
 
-Function CleanUp
```
Comment (philipktlin) [inline scripts/windows/test/Run-E2ETest.ps1]: this will break if someone just call Run-E2Etest.ps1 to run manually on Windows 10 Pro x64.  Please make sure the experience doesn't change as is for manual test run on Windows 10 Pro x64.  And please make it easier to manual run tests for IoT Core.

File: scripts/windows/test/Run-E2ETest.ps1 (hunk 19 above)
Comment (SlavyMihov) [reply scripts/windows/test/Run-E2ETest.ps1]: Got it. Reverted the change.

File: scripts/windows/test/Run-E2ETest.ps1 (hunk 21 above)
Comment (SlavyMihov) [reply scripts/windows/test/Run-E2ETest.ps1]: Why would it break?
The functionality of function CleanUp got moved to Setup-Env.ps1 and the behavior for Windows 10 Pro x64 remains unchanged.

File: scripts/windows/test/Setup-Env.ps1 (hunk 20 above)
Comment (SlavyMihov) [reply scripts/windows/test/Setup-Env.ps1]: Good point, moved.

File: scripts/windows/test/Run-E2ETest.ps1 (hunk 21 above)
Comment (SlavyMihov) [reply scripts/windows/test/Run-E2ETest.ps1]: Why would it break?
The functionality of ClenUp just got moved to Setup-Env.ps1 and remains the same.

File: builds/e2e/longhaul.yaml (hunk 4 above)
Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: Changed to use capabilities _runner-os-name_, _run-stress_ and _run-long-haul_.

File: scripts/windows/test/Run-E2ETest.ps1 (hunk 21 above)
Comment (philipktlin) [reply scripts/windows/test/Run-E2ETest.ps1]: you currently call Setup-Env.ps1 in config yaml.  If I manually run e2e tests using Run-E2ETest.ps1, CleanUp will not be called. right?

Comment (SlavyMihov) [reply scripts/windows/test/Run-E2ETest.ps1]: The _CleanUp_ code used to be called within _Function TestSetup_, which now calls _Setup-Env.ps1_ instead. The functionality of _Setup-Env.ps1_ for Windows Pro x64 remains the same as the former _CleanUp_.

File: scripts/windows/test/Run-E2ETest.ps1 [hunk 22]
```diff
@@ -913,7 +898,7 @@ Function RunTempFilterFunctionsTest
 Function RunTempSensorTest
 {
     PrintHighlightedMessage "Run TempSensor test for $Architecture"
-    TestSetup
+	TestSetup
```
Comment (philipktlin) [inline scripts/windows/test/Run-E2ETest.ps1]: please fix spacing.

Comment (SlavyMihov) [reply scripts/windows/test/Run-E2ETest.ps1]: Fixed.

File: builds/e2e/longhaul.yaml [hunk 23]
```diff
@@ -346,4 +346,69 @@ jobs:
...
+          branchName: $(images.branchName)
+          buildVersionToDownload: latestFromBranch
+          downloadPath: '$(Build.StagingDirectory)'
+          artifactName: $(images.artifact.name.windows)
+          itemPattern: |
+            $(images.artifact.name.windows)/IotEdgeQuickstart/x64/*
+            $(images.artifact.name.windows)/e2e_deployment_files/long_haul_deployment.template.windows.json
+            $(images.artifact.name.windows)/scripts/windows/setup/IotEdgeSecurityDaemon.ps1
+            $(images.artifact.name.windows)/scripts/windows/test/Run-E2ETest.ps1
+            $(images.artifact.name.windows)/scripts/windows/test/Setup-Env.ps1
```
Comment (philipktlin) [inline builds/e2e/longhaul.yaml]: Please update another 2 windows section to download Setup-Env.ps1 file in long haul, stress and e2e tests.

Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: Do you want to do this as part of a separate PR for Windows Pro and Server and keep this one just focused on Windows IoT Core?

Comment (philipktlin) [reply builds/e2e/longhaul.yaml]: Run-E2ETests.ps1 is shared; it will break e2e tests on other Windows Platform, right?

Comment (SlavyMihov) [reply builds/e2e/longhaul.yaml]: Done, added Setup-Env.ps1 to the list of artifact items to be copied for Windows Pro and Server too.
